src/main/java/seedu/address/logic/parser/AddAttendanceCommandParser.java

```diff
@@ -0,0 +1,70 @@
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STUDENT_INDEX;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_WEEK;
+
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.AddAttendanceCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.attendance.Attendance;
+import seedu.address.model.attendance.Week;
+
+/**
+ * Parses input arguments and creates a new AddAttendanceCommand object.
+ */
+public class AddAttendanceCommandParser implements Parser<AddAttendanceCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddAttendanceCommand
+     * and returns an AddAttendanceCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    public AddAttendanceCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_CLASS_INDEX, PREFIX_LESSON_INDEX, PREFIX_STUDENT_INDEX,
+                        PREFIX_WEEK, PREFIX_PARTICIPATION_SCORE);
+
+        Index moduleClassIndex;
+        Index lessonIndex;
+        Index studentIndex;
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_CLASS_INDEX, PREFIX_LESSON_INDEX, PREFIX_STUDENT_INDEX,
+                PREFIX_WEEK, PREFIX_PARTICIPATION_SCORE)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddAttendanceCommand.MESSAGE_USAGE));
+        }
+
+        try {
+            moduleClassIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_CLASS_INDEX).get());
+            lessonIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_LESSON_INDEX).get());
+            studentIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_STUDENT_INDEX).get());
+        } catch (ParseException pe) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddAttendanceCommand.MESSAGE_USAGE), pe);
+        }
+
+        Week week = ParserUtil.parseWeek(argMultimap.getValue(PREFIX_WEEK).get());
+        int participationScore =
+                ParserUtil.parseParticipationScore(argMultimap.getValue(PREFIX_PARTICIPATION_SCORE).get());
+
+        Attendance attendance = new Attendance(participationScore);
+        return new AddAttendanceCommand(moduleClassIndex, lessonIndex, studentIndex, week, attendance);
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
```

Comment:
This seems to appear quite a few times around the code base, could this be placed into another class, perhaps the `ParserUtil`, or any other suitable class and the rest can reference this public method instead.